### PR TITLE
Add user profile pictures

### PR DESCRIPTION
**Pull request by UdaySagar-Git**
## Proposed Changes
- A new bucket for storing profile images of users
- API endpoints to upload/update profile picture of user 

Frontend pr : https://github.com/coronasafe/care_fe/pull/7995

## Merge Checklist

- [ ] Tests added/fixed
- [ ] Update docs in `/docs`
- [ ] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: config/settings/base.py [hunk 1]
```diff
@@ -583,6 +583,19 @@
...
+USER_S3_SECRET = env("USER_S3_SECRET", default=BUCKET_SECRET)
+USER_S3_BUCKET_ENDPOINT = env("USER_S3_BUCKET_ENDPOINT", default=BUCKET_ENDPOINT)
+USER_S3_BUCKET_EXTERNAL_ENDPOINT = env(
+    "USER_S3_BUCKET_EXTERNAL_ENDPOINT",
+    default= BUCKET_EXTERNAL_ENDPOINT
+    if BUCKET_ENDPOINT
+    else USER_S3_BUCKET_ENDPOINT,
+)
+
+
```
Comment (rithviknishad) [inline config/settings/base.py]: Lets re-use the facility bucket itself for user avatars. cc: @sainak

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: config/settings/base.py (hunk 1 above)
Comment (sainak) [reply config/settings/base.py]: @gigincg @vigneshhari

Comment (UdaySagar-Git) [reply config/settings/base.py]: can someone please confirm which bucket to use?

**Review by rithviknishad — APPROVED** [review]

File: care/users/api/viewsets/users.py [hunk 2]
```diff
@@ -366,3 +375,20 @@ def check_availability(self, request, username):
         if User.check_username_exists(username):
             return Response(status=status.HTTP_409_CONFLICT)
         return Response(status=status.HTTP_200_OK)
+
+    @extend_schema(tags=["users"])
+    @action(detail=True, methods=["POST"], permission_classes=[IsAuthenticated])
+    def profile_picture(self, request, username):
+        user = self.get_object()
```
Comment (sainak) [inline care/users/api/viewsets/users.py:405]: role access checks required

Comment (UdaySagar-Git) [reply care/users/api/viewsets/users.py:405]: added , can you please review it once

**Comment by nihal467** [comment]
@UdaySagar-Git can you fix the test and lint failure

**Comment by UdaySagar-Git** [comment]
@nihal467 done

**Review by rithviknishad — APPROVED** [review]

**Comment by rithviknishad** [comment]
@sainak Could you review this?

**Comment by nihal467** [comment]
@UdaySagar-Git the lint is failing and test is failing can you fix it , so we can review the PR

CC: @sainak @rithviknishad @khavinshankar can anyone of you look into the issue as well

File: care/users/api/viewsets/users.py [hunk 3]
```diff
@@ -387,3 +400,40 @@ def check_availability(self, request, username):
...
+            User.TYPE_VALUE_MAP["LocalBodyAdmin"],
+            User.TYPE_VALUE_MAP["DistrictAdmin"],
+            User.TYPE_VALUE_MAP["StateAdmin"],
+        ]
+        return (
+            user.is_superuser
+            or (user.id == request.user.id)
+            or (
+                user.user_type in allowed_user_types
+                and self.facility.has_object_write_permission(request)
```
Comment (sainak) [inline care/users/api/viewsets/users.py]: how can you access self.facility? I think it should be user.home_facility

you also need to check if the request.user's role is higher than the user

**Comment by gigincg** [comment]
Assigning to @sainak to clear issues and close.

File: care/users/api/viewsets/users.py [hunk 4]
```diff
@@ -387,3 +392,28 @@ def check_availability(self, request, username):
...
+
+    @extend_schema(tags=["users"])
+    @profile_picture.mapping.delete
+    def profile_picture_delete(self, request, *args, **kwargs):
+        user = self.get_object()
+        if not self.has_profile_image_write_permission(request, user):
+            return Response(status=status.HTTP_403_FORBIDDEN)
+        user.profile_picture_url = None
+        user.save()
+        return Response(status=status.HTTP_204_NO_CONTENT)
```
Comment (rithviknishad) [inline care/users/api/viewsets/users.py]: Shouldn't we delete the image from the bucket too?

**Review by rithviknishad — APPROVED** [review]

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**